core/src/main/java/com/segment/analytics/kotlin/core/Events.kt

```diff
@@ -86,17 +86,70 @@ sealed class BaseEvent {
     }
 
     internal suspend fun applyBaseEventData(store: Store) {
-        val system = store.currentState(System::class) ?: return
         val userInfo = store.currentState(UserInfo::class) ?: return
 
         this.anonymousId = userInfo.anonymousId
-        this.integrations = system.integrations ?: emptyJsonObject
+        this.integrations = emptyJsonObject
 
         if (this.userId.isBlank()) {
             // attach system userId if present
             this.userId = userInfo.userId ?: ""
         }
     }
+
+    // Create a shallow copy of this event payload
+    fun <T : BaseEvent> copy(): T {
```

Comment:
can we make this method abstract, let the subclasses implement the copy details? that case, the parent class doesn't have to know it's children. if we add more subclasses, we don't need to modify the parent class.

Reply:
i chose to do this in the BaseEvent class for a 2 reasons
- We know that we wont have more than track/identify/group/screen/alias event types
- BaseEvent is a sealed type so its ok to have the full implementation in the parent class (which knows about all the subclasses)

Reply:
that makes sense